File: Open3D/09_icp_color/python/main.py
import open3d as o3d, numpy as np
s=o3d.io.read_point_cloud("source.ply"); t=o3d.io.read_point_cloud("target.ply");
for p in (s,t): p.estimate_normals(o3d.geometry.KDTreeSearchParamHybrid(radius=0.05,max_nn=30))
r=o3d.pipelines.registration.registration_colored_icp(s,t,0.05,np.eye(4)); print(r.transformation)

File: Open3D/09_icp_color/cpp/main.cpp
#include <open3d/Open3D.h>
#include <iostream>
int main(){ auto s=open3d::io::CreatePointCloudFromFile("source.ply"), t=open3d::io::CreatePointCloudFromFile("target.ply"); if(!s||!t) return 1; s->EstimateNormals(open3d::geometry::KDTreeSearchParamHybrid(0.05,30)); t->EstimateNormals(open3d::geometry::KDTreeSearchParamHybrid(0.05,30)); auto r=open3d::pipelines::registration::RegistrationColoredICP(*s,*t,0.05,Eigen::Matrix4d::Identity()); std::cout<<r.transformation_<<"\n"; }
